hooks/useFetchStakingPoolsData.js

```diff
@@ -0,0 +1,239 @@
+        .method(accountBalanceOfABI),
+
+      // Unstaked staking token balance
+      getUnstakedBalanceOf: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(balanceOfABI),
+
+      // Unstaked staking token approval amount
+      getUnstakedAllowanceAmount: connex?.thor
+        .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+        .method(allowanceABI),
+
+      // Claimable token
+      getEarned: connex?.thor
+        .account(stakingPool.rewardsAddress[VECHAIN_NODE])
+        .method(earnedABI),
+    };
+  });
+
+  const getPoolData = async () => {
+    return await Promise.all(
+      STAKING_POOLS.map(async (stakingPool) => {
+        // Pool size
+        const {
+          decoded: { 0: poolSize },
+        } = await stakingPoolsFunctions[stakingPool.id].getBalanceOf.call();
+
+        const rewardDataABI = find(MultiRewards, { name: "rewardData" });
+        let method =
+          connexStakingPools[stakingPool.id].rewardsContract.method(
+            rewardDataABI
+          );
+        let res = await method.call(stakingPool.rewardTokens[0].address[VECHAIN_NODE]);
+        const rewardRate = ethers.BigNumber.from(res.decoded.rewardRate);
+
+        const totalSupplyABI = find(IERC20, { name: "totalSupply" });
+        method = connex.thor
+          .account(stakingPool.stakingTokenAddress[VECHAIN_NODE])
+          .method(totalSupplyABI);
+        res = await method.call();
+
+        const totalLPTokenSupply = BigNumber.from(res.decoded[0]);
+        const numberOfLPTokensStaked = poolSize;
+        // This is calculated in the amount of vex tokens instead of usd
+        const tokenTvl =
+          poolInfo[stakingPool.id].pair.tokenAmounts[0].toFixed(18);
+        // Multiply by two to get the pool TVL as this is a 50-50 pool
+        const tvlInToken = parseUnits(tokenTvl, 18).mul(2);
+
+        try {
+          const apr = rewardRate
+            .mul(NUM_SECONDS_IN_A_YEAR)
+            .mul(parseUnits("1", "ether")) // More units for precision
+            .mul(100) // Convert into percentage
+            // The following two lines take into account
+            // that not all LP tokens are staked on the staking site
+            .mul(totalLPTokenSupply)
+            .div(numberOfLPTokensStaked)
+            // We divide all the rewards by the total VEX TVL
+            .div(tvlInToken);
+
+          const tvlInUsd = tvlInToken
+            .mul(parseUnits(poolInfo[stakingPool.id].usdPerToken.toString()))
+            .mul(numberOfLPTokensStaked)
+            .div(totalLPTokenSupply)
+            .div(parseUnits("1", "ether"));
+
+          return {
+            poolId: stakingPool.id,
+            vault: stakingPool.stakeAsset,
+            poolSize: BigNumber.from(poolSize),
+            apr,
+            tvlInUsd,
+          };
+        } catch (err) {
+          console.log(`Pool ${stakingPool.id} is empty`)
```

Comment:
we don't log the error anywhere?

Reply:
this is more a warning than an error, we can use Sentry or something else, what do you suggest?

Reply:
Just `console.log(err);` unless we are confident it will always be because of empty staking pool? I wouldn't worry too much about users seeing the errors in console